src/extensibility/ExtensionManagerDialog.js

```diff
@@ -107,57 +110,98 @@ define(function (require, exports, module) {
                     .text(Strings.PROCESSING_EXTENSIONS)
                     .append("<span class='spinner inline spin'/>");
                 
-                ExtensionManager.removeMarkedExtensions()
+                var removeExtensionsPromise,
+                    updateExtensionsPromise,
+                    disableExtensionsPromise,
+                    removeErrors,
+                    updateErrors,
+                    disableErrors;
+                
+                removeExtensionsPromise = ExtensionManager.removeMarkedExtensions();
+                removeExtensionsPromise
+                    .fail(function (errorArray) {
+                        removeErrors = errorArray;
```

Comment:
I think that the code might look better if you parsed the error array here and created the dialog, instead of doing it in the general fail.